sdk/src/main/java/com/horizen/account/api/rpc/request/RpcBatchRequest.java

```diff
@@ -0,0 +1,54 @@
+package com.horizen.account.api.rpc.request;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.horizen.account.api.rpc.handler.RpcException;
+import com.horizen.account.api.rpc.utils.RpcCode;
+import com.horizen.account.api.rpc.utils.RpcError;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * [
+ * {"id":1648039192785,"jsonrpc":"2.0","method":"eth_chainId","params":[]},
+ * {"id":3918473928749,"jsonrpc":"2.0","method":"eth_blockNumber","params":[]}
+ * ]
+ */
+public class RpcBatchRequest {
+
+    private List<RpcRequest> rpcRequests;
+    private int invalidRequestsNumber;
+
+    public RpcBatchRequest(JsonNode json) throws RpcException {
+
+        if (json.isArray() && json.isEmpty()) {
+            throw new RpcException(RpcError.fromCode(RpcCode.ParseError));
+        }
+
+        List<RpcRequest> inputRpcRequests = new ArrayList<>();
+        for(JsonNode jsonItem: json) {
+            try {
+                RpcRequest rpcRequest = new RpcRequest(jsonItem);
+                inputRpcRequests.add(rpcRequest);
+            } catch (RpcException e) {
+                invalidRequestsNumber++;
+            }
```

Comment:
I think we should detect that the JSON is an array and keep entries as json object elements of that array.
It will allow us to properly process single requests and properly return errors (in case of invalid request, for example)

Reply:
Agree on this point, following the refactor the class `RpcBatchRequet` was eliminated from the project